patches/squirrel.mac/feat_add_new_squirrel_mac_bundle_installation_method_behind_flag.patch

```diff
@@ -0,0 +1,128 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <samuel.r.attard@gmail.com>
+Date: Mon, 28 Mar 2022 02:43:18 -0700
+Subject: feat: add new Squirrel.Mac bundle installation method behind flag
+
+The 'SquirrelMacEnableDirectContentsWrite' user default in your apps defaults suite
+can be used to control this new installation method.  It is designed to remove the
+requirement that the updating process have write access to its parent directory.
+
+With this feature enabled the updating process only needs write access to it's own
+.app bundle folder, not the owning /Applications folder.  This should allow more
+non-admin users to update applications when appropriately granted group based permissions.
+
+E.g. 775 and chown :staff
+
+diff --git a/Squirrel/SQRLInstaller.m b/Squirrel/SQRLInstaller.m
+index 7dd98ddee4ae0f4e01fd7aaa3486083bff7d0da1..c1f328fa8c3689218ef260347cb8f9d30b789efe 100644
+--- a/Squirrel/SQRLInstaller.m
++++ b/Squirrel/SQRLInstaller.m
+@@ -249,6 +249,7 @@ - (RACSignal *)acquireTargetBundleURLForRequest:(SQRLShipItRequest *)request {
+ 		] reduce:^(NSURL *directoryURL, SQRLCodeSignature *codeSignature) {
+ 			NSURL *targetBundleURL = request.targetBundleURL;
+ 			NSURL *newBundleURL = [directoryURL URLByAppendingPathComponent:targetBundleURL.lastPathComponent];
++			[NSFileManager.defaultManager createDirectoryAtURL:newBundleURL withIntermediateDirectories:FALSE attributes:nil error:nil];
```

Comment:
Do we need to catch the situation in which the directory creation fails?

Reply:
If the directory creation fails it will fall back to the old `.app` moving strategy, it's theoretically impossible for this directory creation to fail though as it's making a directory in an already owned and created temp folder.